frontend/app/src/components/helper/PrioritizedListEntry.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <v-row align="center">
+    <v-col v-if="data.icon" cols="auto">
+      <adaptive-wrapper>
+        <v-img
+          :width="size"
+          contain
+          position="left"
+          :max-height="size"
+          :src="data.icon"
+        />
+      </adaptive-wrapper>
+    </v-col>
+    <v-col v-if="data.identifier === PriceOracle.UNISWAP2" cols="auto">
+      {{ t('oracles.uniswap_v2') }}
```

Comment:
That would look much nicer. The problem I get here is that `data.identifier` has type string and the keys of the `labels` object have the values of the **two** enum types and I get the following error:

`Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ uniswapv2: TranslateResult; uniswapv3: TranslateResult; manualcurrent: TranslateResult; blockchain_account: TranslateResult; ... 4 more ...; private_addressbook: TranslateResult; }'.
  No index signature with a parameter of type 'string' was found on type '{ uniswapv2: TranslateResult; uniswapv3: TranslateResult; manualcurrent: TranslateResult; blockchain_account: TranslateResult; ... 4 more ...; private_addressbook: TranslateResult; }'.`

My preferred solution was to put these translated strings into the `PrioritizedListItemData` class/interface as a property `displayName` for example. This property would be initialized along with the identifier and other stuff like the icons. In the  `PrioritizedListEntry` there is only a `tr(data.displayName)`. This would have the benefit, that this visual item data is defined in one place and the generic `PrioritizedListEntry` component is not aware of the implementations and their translation strings. 
Unfortunately this this result in linter errors complaining about missing translations. I guess the linter is only looking in Vue components and not in .ts scripts. I haven't found a good solution for this after some investigation.

Do you have any solution for this? I must admit, that I'm not that familiar with typescript. I'm more a backend dev (go, C++, a bit of python ;) ) and working with frontend (Angular) for some months now.

Reply:
Let me think a bit about it and look at the code again.

Does `data.identifier` need to be a string? It can only take values from the two enums right?

Reply:
> Let me think a bit about it and look at the code again.
> 
> Does `data.identifier` need to be a string? It can only take values from the two enums right?

Ok I see that it can be more, so what I would do in this case I would go the generics way

```typescript
export class PrioritizedListData<T = string> {
  constructor(private itemData: Array<PrioritizedListItemData<T>>) {}

  itemIdsNotIn(itemIds: T[]): T[] {
    return this.itemData
      .filter(item => !itemIds.includes(item.identifier))
      .map(itemData => itemData.identifier);
  }

  itemDataForId(id: T): PrioritizedListItemData<T> | undefined {
    return this.itemData.find(item => item.identifier === id);
  }
}

export interface PrioritizedListItemData<T = string> {
  identifier: T;
  icon?: string;
  extraDisplaySize?: string; // may be a string specifying pixels e.g. 12px
}
```

and then at the data initialisation I would pass the enum types so that identifier is correctly typed.

```typescript
const props = defineProps({
  data: { required: true, type: Object as PropType<PrioritizedListItemData<AddressNamePriority>> }
});
```

Then use a mapped type for the labels:

```typescript
const labels: { [keys in AddressNamePriority]: string } = {
  [AddressNamePriority.BLOCKCHAIN_ACCOUNT]: t(
    'eth_address_book.hint.priority.list.private_address_book'
  ).toString()
  ...
};
```

This way `{{ labels[data.identifier] }}` should work without any issues.

Reply:
Maybe I get something wrong here, but in this case I need to specify the `PrioritizedList.vue` and the `PrioritizedListEntry.vue` twice, once with `AddressNamePriority` type and once with `PriceOracle` type? 
The labels object won't take that `PriceOracle` enum anymore :(

Reply:
You no need you can a custom type that is `AddressNamePriority | PriceOracle`

And if you only need specific types from `PriceOracle` you can use [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) to narrow down the type. This should include all the properties you want.

Reply:
If it is overly complicated let me know when you are don't with other review comments and I can take it over and push on top of your PR.

Reply:
I'll try it by myself and will come back to you if I'm stuck. Thanks for your help!